packages/pocket-ic/tests/tests.rs

```
@@ -946,53 +946,99 @@ fn test_schnorr() {
+                                )
+                                .unwrap();
+                                let merkle_root =
+                                    bitcoin::util::taproot::TapBranchHash::from_slice(
+                                        &bip341_aux.merkle_root_hash,
+                                    )
+                                    .unwrap();
+                                let secp256k1_engine = bitcoin::secp256k1::Secp256k1::new();
+                                xonly
+                                    .tap_tweak(&secp256k1_engine, Some(merkle_root))
+                                    .0
+                                    .to_inner()
+                                    .serialize()
+                                    .to_vec()
+                            }
+                        };
+                        let vk = VerifyingKey::from_bytes(&public_key).unwrap();
+                        let sig = Signature::try_from(schnorr_signature_result.unwrap().as_slice())
+                            .unwrap();
+
+                        vk.verify_prehash(message, &sig).unwrap();
+                    }
+                    SchnorrAlgorithm::Ed25519 => {
+                        use ed25519_dalek::{Signature, Verifier, VerifyingKey};
+                        let pk: [u8; 32] = schnorr_public_key.public_key.try_into().unwrap();
+                        let vk = VerifyingKey::from_bytes(&pk).unwrap();
+                        let verification_result = schnorr_signature_result.and_then(|signature| {
+                            {
+                                let s = Signature::from_slice(&signature).unwrap();
+                                vk.verify(message, &s)
+                            }
+                            .map_err(|e| format!("{:?}", e))
+                        });
+                        assert!(
+                            verification_result.is_ok() == aux.is_none(),
```

Is it expected that `verification_result` could be `Err(...)` because of `schnorr_signature_result` or `vk.verify`? Based on that, let's please unwrap in the case that is not expected to fail.

---

Do you mean ignore the error case?

---

Ah, yes, that's true. We only need to replace `and_then` by `map`. Done.